pkg/virtctl/console/console.go

```diff
@@ -26,56 +26,64 @@ import (
 	"os"
 	"os/signal"
 
-	flag "github.com/spf13/pflag"
 	"golang.org/x/crypto/ssh/terminal"
-	"k8s.io/api/core/v1"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/client-go/tools/clientcmd"
 
 	"kubevirt.io/kubevirt/pkg/kubecli"
-	"kubevirt.io/kubevirt/pkg/virtctl"
 )
 
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	return &cobra.Command{
+		Use:   "console (vm)",
+		Short: "Connect to a console of a virtual machine.",
+		Long:  usage(),
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Console{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+}
+
 type Console struct {
+	clientConfig clientcmd.ClientConfig
 }
 
-func (c *Console) FlagSet() *flag.FlagSet {
-	cf := flag.NewFlagSet("console", flag.ExitOnError)
+func (c *Console) FlagSet() *pflag.FlagSet {
+	cf := pflag.NewFlagSet("console", pflag.ExitOnError)
 
 	return cf
 }
 
-func (c *Console) Usage() string {
-	usage := "Connect to a serial console on a VM:\n\n"
+func usage() string {
+	usage := "Connect to a console of a virtual machine.\n\n"
 	usage += "Examples:\n"
 	usage += "# Connect to the console on VM 'myvm':\n"
 	usage += "virtctl console myvm\n\n"
-	usage += "Options:\n"
-	usage += c.FlagSet().FlagUsages()
 	return usage
 }
 
-func (c *Console) Run(flags *flag.FlagSet) int {
-	server, _ := flags.GetString("server")
-	kubeconfig, _ := flags.GetString("kubeconfig")
-	namespace, _ := flags.GetString("namespace")
-	if namespace == "" {
-		namespace = v1.NamespaceDefault
-	}
-	if len(flags.Args()) != 2 {
-		log.Println("VM name is missing")
-		return virtctl.STATUS_ERROR
+func (c *Console) Run(cmd *cobra.Command, args []string) error {
+	kubecli.GetKubevirtClientConfig()
```

Comment:
Why are the return values not set/used here?  It's not obvious to me why GetKubevirtClientConfig is called here.

Reply:
Done.